agent/ae/ae_test.go

```diff
@@ -392,7 +391,12 @@ func (m *mock) SyncChanges() error {
 }
 
 func testSyncer() *StateSyncer {
-	logger := log.New(os.Stderr, "", 0)
+	consulLogger := hclog.New(&hclog.LoggerOptions{
+		Level: 0,
```

Comment:
To carry the prior behavior through explicitly, probably this should still set `os.Stderr` even if that's the default. Though it would be better if we could pass a `*testing.T` or `testing.TB` in to the `testSyncer` function so that a `testutil.TestWriter` could be used instead of `Stderr`.

Reply:
This is called only 23 times in one file, I could change it so we pass `*testing.T` in through the function so it can go out `testutil.TestWriter`
I like this suggestion to pretty-ify things more.